tests/compiletest.rs

```diff
@@ -274,10 +274,16 @@ fn main() -> Result<()> {
 fn run_dep_mode(target: String, mut args: impl Iterator<Item = OsString>) -> Result<()> {
     let path = args.next().expect("./miri run-dep must be followed by a file name");
     let mut config = test_config(&target, "", Mode::Yolo, /* with dependencies */ true);
+    config.build_dependencies_and_link_them()?;
     config.program.args.remove(0); // remove the `--error-format=json` argument
-    config.program.args.push("--color".into());
-    config.program.args.push("always".into());
-    let mut cmd = ui_test::test_command(config, Path::new(&path))?;
+    let mut cmd = config.program.build(&config.out_dir);
+
+    cmd.arg("--color").arg("always");
+    cmd.arg(path);
+
+    if let Some(edition) = config.edition {
+        cmd.arg("--edition").arg(edition);
+    }
```

Comment:
Does this mean `./miri run-dep`  defaults to edition 2021? Should we make `./miri run` also add `--edition 2021`?

Reply:
probably 😆 Seems like a good default to have

Reply:
But it's then hard to overwrite... so maybe let's just remove it from run-dep

Reply:
Will `./miri run-dep file.rs --edition 2021` work?

Reply:
It works now 😁 I tested that the behaviour around `--` is now the same as with `./miri run`.